evalml/model_understanding/_partial_dependence_fast_mode_utils.py

```diff
@@ -0,0 +1,103 @@
+import pandas as pd
+
+from evalml.utils.woodwork_utils import infer_feature_types
+
+
+def _get_cloned_feature_pipelines(
+    features,
+    pipeline,
+    variable_has_features_passed_to_estimator,
+    X_train,
+    y_train,
+):
+    """Clones and fits pipelines for partial dependence fast mode.
+
+    Args:
+        pipeline (PipelineBase or subclass): Fitted pipeline that will be cloned.
+        features (string, tuple[int or string]): The target feature for which to create the partial dependence plot for.
+            If features is a string, it must be a valid column name in X.
+            If features is a tuple of int/strings, it must contain valid column integers/names in X.
+        variable_has_features_passed_to_estimator (dict[str, bool]): Dictionary mapping variable names to whether or not
+            the variable has an impact on predictions. Variables that do not have features passed to the estimator
+            will not have an impact on predictions and therefore do not need to have pipelines fitted for them.
+        X_train (pd.DataFrame, np.ndarray): The data that was used to train the original pipeline. Will
+            be used to train the cloned pipelines.
+        y_train (pd.Series, np.ndarray): The target data that was used to train the original pipeline. Will
+            be used to train the cloned pipelines.
+
+    Returns:
+        dict[str, PipelineBase or subclass]: Dictionary mapping feature name to the pipeline pipeline
+            fit for it.
+    """
+    if X_train is None or y_train is None:
+        raise ValueError("Training data is required for partial dependence fast mode.")
```

Comment:
This is purely my opinion, but it may be neater to raise this error higher up in the call stack - potentially even in the highest level `partial_dependence` function. Just as fewer things to wade through when debugging an error.

Reply:
I think that makes sense--moved up to `partial_dependence` with all the other error logic